server/src/main/java/org/opensearch/action/admin/indices/settings/put/TransportUpdateSettingsAction.java

```diff
@@ -64,6 +70,18 @@ public class TransportUpdateSettingsAction extends TransportClusterManagerNodeAc
 
     private static final Logger logger = LogManager.getLogger(TransportUpdateSettingsAction.class);
 
+    private final static String[] ALLOWLIST_REMOTE_SNAPSHOT_SETTINGS = {
+        "index.max_result_window",
+        "index.max_inner_result_window",
+        "index.max_rescore_window",
+        "index.max_docvalue_fields_search",
+        "index.max_script_fields",
+        "index.max_terms_count",
+        "index.max_regex_length",
+        "index.highlight.max_analyzed_offset" };
+
+    private final static String[] ALLOWLIST_REMOTE_SNAPSHOT_SETTINGS_PREFIXES = { "index.search.slowlog" };
```

Comment:
Nitpick, but for both of these I'd use `Set.of(...)` to create immutable sets. It'll save you the step of doing `Arrays.asList` below.